framework/tests/packets.rs

```diff
@@ -0,0 +1,318 @@
+        let seg0 = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7334").unwrap();
+        let seg1 = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7335").unwrap();
+        assert_eq!(srh.next_header().unwrap(), NextHeader::Udp);
+        assert_eq!(srh.hdr_ext_len(), 4);
+        assert_eq!(srh.routing_type(), 4);
+        assert_eq!(srh.segments_left(), 0);
+        assert_eq!(srh.last_entry(), 1);
+        assert_eq!(srh.protected(), false);
+        assert_eq!(srh.oam(), false);
+        assert_eq!(srh.alert(), false);
+        assert_eq!(srh.hmac(), false);
+        assert_eq!(srh.tag(), 0);
+        assert_eq!(srh.segments().len(), 2);
+        assert_eq!(srh.segments()[0], seg0);
+        assert_eq!(srh.segments()[1], seg1);
+    }
+}
+
+fn insert_static_srh_from_bytes() {
+    let packet_header = [
+        // --- Ethernet header ---
+        // Destination MAC
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x01,
+        // Source MAC
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x02,
+        // EtherType (IPv6)
+        0x86,
+        0xDD,
+        // --- IPv6 Header ---
+        // Version, Traffic Class, Flow Label
+        0x60,
+        0x00,
+        0x00,
+        0x00,
+        // Payload Length
+        0x00,
+        0x18,
+        // Next Header (UDP = 17)
+        0x11,
+        // Hop Limit
+        0x02,
+        // Source Address
+        0x20,
+        0x01,
+        0x0d,
+        0xb8,
+        0x85,
+        0xa3,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x01,
+        // Dest Address
+        0x20,
+        0x01,
+        0x0d,
+        0xb8,
+        0x85,
+        0xa3,
+        0x00,
+        0x00,
+        0x00,
+        0x00,
+        0x8a,
+        0x2e,
+        0x03,
+        0x70,
+        0x73,
+        0x34,
+    ];
+    let pkt = packet_from_bytes(&packet_header);
+    let epkt = pkt.parse_header::<MacHeader>();
+    let mut v6pkt = epkt.parse_header::<Ipv6Header>();
+    let mut v6pkt2 = v6pkt.clone();
+    let v6h = v6pkt.get_mut_header();
+    {
+        assert_eq!(v6h.next_header().unwrap(), NextHeader::Udp);
+        v6h.set_next_header(NextHeader::Routing);
+        assert_eq!(v6h.next_header().unwrap(), NextHeader::Routing);
+    }
+
+    assert_eq!(v6h.peek_num_segments(), None);
+    let seg0 = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7335").unwrap();
+    let seg1 = Ipv6Addr::from_str("2001:db8:85a3::8a2e:0370:7336").unwrap();
+    let mut srh = <SegmentRoutingHeader<Ipv6Header, U2>>::new(arr![Segment; seg0, seg1]);
+    {
+        srh.set_routing_type(ROUTING_TYPE);
+        srh.set_segments_left(0);
+        srh.set_last_entry(1);
+
+        srh.ext_header.set_next_header(v6h.next_header().unwrap());
+        HdrExtLen::set_hdr_ext_len(&mut srh.ext_header, 2);
+        assert_eq!(srh.ext_header.hdr_ext_len(), 4);
```

Comment:
This is what I was saying earlier about the trait fn being confusing. You have `set_hdr_ext_len` that gets passed in `2`, but a getter that's similarly named that returns `4`.

Reply:
i was testing both versions, but i'll rename it.